pkg/interface.go

```diff
@@ -62,6 +62,9 @@ type Client interface {
 	// If there's no TLS certificate for the given entity nil is returned. In that case RenewTLSCertificate can be used
 	// to (re)issue the certificate.
 	GetTLSCertificate(entity types.LegalEntity) (*x509.Certificate, crypto.PrivateKey, error)
+	// SignTLSCertificate creates a CSR for the given public key and will sign it with the vendorCA.
+	// It differs from GetTLSCertificate in the fact that the private key is unknown/external.
+	SignTLSCertificate(publicKey crypto.PublicKey) (*x509.Certificate, error)
```

Comment:
maybe `GetTLSCertificate` and `RenewTLSCertificate` should be marked deprecated in favor of the new `SignTLSCertificate`. They're only used by `nuts-network` which should imo use the same TLS certificate as `SignTLSCertificate` issues?

Reply:
no, the 3 will eventually use some of the same stuff, but this one does not allow the private key to be internal.